.github/scripts/apply_shellcheck.sh

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+set -eu
+
+apply_shellcheck() {
+  local filename=${1:?}
+  local code=${2:-""}
+  patch="/tmp/patch.${RANDOM}"  # Do not use PWD, as this will create tons of patch files
+  [[ -f "${patch}" ]] && rm -f "${patch}"
+  set +e
+  shellcheck -i "${code}" "${filename}" -f diff > "${patch}"
+  set -e
+  # shellcheck disable=SC2312
+  # Ignore the error that masks the return value
+  patch_size=$(wc -l "${patch}" | awk '{print $1}')
+  # shellcheck disable=
+  if [[ "${patch_size}" -gt 0 ]]; then
+    echo "Apply patch for ${code} on ${filename}"
+    git apply "${patch}"
+  else
+    echo "No patch to apply for ${code} on ${filename}"
+  fi
+  rm -f "${patch}"
+}
+
+file=${1:?}  # File to fix shellcheck errors
+apply_shellcheck "${file}" SC2292  # Prefer [[ ]] over [ ] for tests in Bash/Ksh
+apply_shellcheck "${file}" SC2250  # Prefer putting braces around variable e.g. ${variable}
+apply_shellcheck "${file}" SC2248  # Prefer double quoting
+apply_shellcheck "${file}" SC2086  # Double quote to prevent globbing and word splitting
```

Comment:
This seems unrelated to this PR and would best be served in a separate PR.

I am also pretty leery of auto-applying shellcheck patches. The fixes aren't always great (like non-canonical quotations), plus there are other more appropriate tools to do it (like `shfmt`)

Reply:
Feel free to add a tool that makes it easier on the user to fix these linter errors.

Reply:
This file is now removed from the PR